### PR TITLE
Allows lev.py to be imported as well as invoked

### PR DESCRIPTION
**Pull request by harikv**
- Uses __name__ == __main__ check which was introduced in py3
- Modified levenshtein() and digit_compare() to have an
additional parameter `distance` to allow it to tweaked while
importing as well
- Added .gitignore to ignore pycache folders which can be created
while importing from a git submodule

### CODE REVIEW

**Comment by harikv** [comment]
@tobylunt This change will mean that masala-merge cannot be used with < Python3.0. I've used this method to make it easier to modify/maintain the library in future since you don't have to worry about multiple files. If lack of Python2.x support is a dealbreaker I can do it in a slightly different manner (creating another file which cannot be invoked from the CLI to allow that to be imported).

File: lev.py [hunk 1]
```diff
@@ -176,7 +176,8 @@ def update_matrix(matrix, row, col, value):
     if matrix[row][col] > value: matrix[row][col] = value
 
 # internet levensthein
-def levenshtein(str1, str2):
+# Added distance as optional parameter to avoid breaking function when imported
+def levenshtein(str1, str2, distance = 1000):
```
Comment (harikv) [inline lev.py:191]: @tobylunt This is the key breaking change, as previously we were assuming the script will only be invoked from the CLI. Let me know if this is alright. I've ensured that all invocations of the levenshtein() function either have options.distance or a default passed as a parameter.

**Comment by tobylunt** [comment]
Looks good Hari, thanks!

**Comment by tobylunt** [comment]
Looks good Hari, thanks!

**Pull request merged**